flower_shop/models/stockLot.py

```
@@ -0,0 +1,36 @@
+from odoo import models,fields,api
+
+class StockProduction(models.Model):
+   _inherit="stock.lot"
+
+   water_ids=fields.One2many("flower.water","serial_id")
+   is_flower= fields.Boolean(related='product_id.is_flower')
+
+
+   def action_water_flower(self):
+      for recod in self.filtered(lambda rec: rec.is_flower):    
+         if record.water_ids:
+            last_watered_date = record.water_ids[0].date
+            frequency = record.product_id.flower_id.watering_frequency
+            today = fields.Date.today()
+            days_since_last_watering = (today - last_watered_date ).days
+            if (today - last_watered_date).days < frequency:
+                 self.env["flower.water"].create({
+                  "serial_id": record.id,
+             })
```

this should throw an error if the condition is true and create if the condition is false